src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -0,0 +1,342 @@
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj.util.Color8Bit;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.button.Trigger;
+import frc.robot.Constants.CANIDs;
+import frc.robot.Constants.DIOConstants;
+import frc.robot.Robot;
+import utilities.DebugEntry;
+import utilities.TOFSensorSimple;
+import utilities.TOFSensorSimple.TOFType;
+
+public class IntakeSubsystem extends SubsystemBase {
+  /** Creates a new IntakeSubsystem. */
+  private TalonFX intakeMotor;
+
+  private TalonFXSimState simPivotMotor;
+
+  private TalonFX pivotMotor;
+  private TalonFX conveyorMotor;
+  private Angle pivotSetpoint = kPivotRetractAngle;
+
+  private DutyCycleEncoder throughBoreEncoder;
+
+  // Sensor closest to birdhouse
+  private TOFSensorSimple sensor;
+  private SimDeviceSim simSensor;
+  private SimBoolean simbeam;
+
+  private Mechanism2d mech = new Mechanism2d(2, 2);
+  private ComplexWidget widget;
+  private MechanismLigament2d pivotArmMech;
+
+  private enum IntakeState {
+    Idle,
+    atSensor,
+    coralLoaded
+  }
+
+  private IntakeState state = IntakeState.Idle;
+  private Timer timer = new Timer();
+
+  private SingleJointedArmSim pivotSim;
+
+  private DebugEntry<Double> pivotOutput;
+  private DebugEntry<String> currentCommand;
+
+  public IntakeSubsystem() {
+    intakeMotor = new TalonFX(CANIDs.kIntakeMotor);
+    pivotMotor = new TalonFX(CANIDs.kPivotMotor);
+    conveyorMotor = new TalonFX(CANIDs.kConveyorMotor);
+    sensor = new TOFSensorSimple(CANIDs.kConveyorSensor, Inches.of(1), TOFType.LASER_CAN);
+    throughBoreEncoder = new DutyCycleEncoder(DIOConstants.kthroughBoreEncoderID, 1, armZero);
+
+    var slot0Configs = new Slot0Configs();
+    slot0Configs.kP = kPivotP;
+    slot0Configs.kI = kPivotI;
+    slot0Configs.kD = kPivotD;
+    slot0Configs.kG = kPivotG;
+    slot0Configs.kA = kPivotA;
+    slot0Configs.kV = kPivotV;
+    slot0Configs.GravityType = kPivotGravityType;
+
+    var feedbackConfigs = new FeedbackConfigs();
+    feedbackConfigs.RotorToSensorRatio = 1;
+    feedbackConfigs.SensorToMechanismRatio = kSensorToMechanism;
+
+    var pivotMotionMagic =
+        new MotionMagicConfigs()
+            .withMotionMagicCruiseVelocity(kMotionMagicCruiseVelocity)
+            .withMotionMagicAcceleration(kMotionMagicAcceleration)
+            .withMotionMagicJerk(kMotionMagicJerk);
+
+    pivotMotor.getConfigurator().apply(slot0Configs);
+    pivotMotor.getConfigurator().apply(feedbackConfigs);
+    pivotMotor.getConfigurator().apply(pivotMotionMagic);
+    pivotMotor.setControl(new CoastOut()); // Temporary
+
+    pivotOutput = new DebugEntry<Double>(0.0, "Pivot Output", this);
+    currentCommand = new DebugEntry<String>("none", "Pivot Command", this);
+
+    pivotMotor.setPosition(throughBoreEncoder.get());
```

Comment:
Remove this so that we rely on the button press to initialize the absolute position.